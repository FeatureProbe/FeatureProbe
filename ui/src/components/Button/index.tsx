import { useRef, MouseEvent, useCallback } from 'react';
import { Button, ButtonProps } from 'semantic-ui-react';

const CustomButton = ({children, ...props}: ButtonProps) => {
  const { onClick } = props;
  const buttonRef = useRef(null);
  
  const handleOnClick = useCallback((e: MouseEvent<HTMLButtonElement, globalThis.MouseEvent>, {...props}) => {
    // @ts-ignore object null compatibility
    buttonRef.current.ref?.current.blur();

    if (onClick) {
      onClick(e, props);
    }
  }, [onClick]);

  return (
    <Button {...props} ref={buttonRef} onClick={handleOnClick}>
      { children }
    </Button>
  );
};

export default CustomButton;