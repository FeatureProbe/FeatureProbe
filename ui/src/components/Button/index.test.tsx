import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import Button from '.';

it('button snapshot', () => {
  (async () => {
    const { asFragment } = render(<Button>this is a Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  })();
});

test('click button', (done) => {
  (async () => {
    const { rerender } = render(<Button key={1} onClick={(e) => {e.currentTarget.innerHTML = 'you clicked the button';}}>this is a Button</Button>);
    let html = screen.getByRole('button');
    await userEvent.click(html);
    expect(html.innerHTML).toBe('you clicked the button');

    rerender(<Button key={2}>this is a Button</Button>);
    html = screen.getByRole('button');
    await userEvent.click(html);
    expect(html.innerHTML).toBe('this is a Button');

    done();
  })();
});
