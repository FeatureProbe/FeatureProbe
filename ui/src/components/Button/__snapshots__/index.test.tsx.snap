// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshot 1`] = `
<DocumentFragment>
  <button
    class="ui button"
  >
    this is a Button
  </button>
</DocumentFragment>
`;
